lib/segment/src/segment/entry.rs

```diff
@@ -879,25 +964,108 @@ impl SegmentEntry for Segment {
     }
 }
 
+impl Segment {
+    fn files(&self) -> Vec<PathBuf> {
+        let mut files = Vec::new();
+
+        for vector_data in self.vector_data.values() {
+            files.extend(vector_data.vector_index.borrow().files());
+            files.extend(vector_data.vector_storage.borrow().files());
+
+            if let Some(quantized_vectors) = vector_data.quantized_vectors.borrow().deref() {
+                files.extend(quantized_vectors.files());
+            }
+        }
+
+        files.extend(self.payload_index.borrow().files());
+        files.extend(self.payload_storage.borrow().files());
+
+        files.extend(self.id_tracker.borrow().files());
+
+        files
+    }
+
+    fn versioned_files(&self) -> Vec<(PathBuf, u64)> {
+        let mut files = Vec::new();
+
+        for vector_data in self.vector_data.values() {
+            files.extend(vector_data.vector_index.borrow().versioned_files());
+            files.extend(vector_data.vector_storage.borrow().versioned_files());
+
+            if let Some(quantized_vectors) = vector_data.quantized_vectors.borrow().deref() {
+                files.extend(quantized_vectors.versioned_files());
+            }
+        }
+
+        files.extend(self.payload_index.borrow().versioned_files());
+        files.extend(self.payload_storage.borrow().versioned_files());
+
+        files.extend(self.id_tracker.borrow().versioned_files());
+
+        files
+    }
+}
+
+fn updated_files(old: &SegmentManifest, current: &SegmentManifest) -> HashSet<PathBuf> {
+    // Compare two segment manifests, and return a list of files from `current` manifest, that
+    // should be included into partial snapshot.
+
+    let mut updated = HashSet::new();
+
+    for (path, &current_version) in &current.files {
+        // Include file into partial snapshot if:
+        //
+        // 1. `old` manifest does not contain this file
+        let Some(old_version) = old.files.get(path).copied() else {
+            updated.insert(path.clone());
+            continue;
+        };
+
+        // 2. if `old` manifest contains this file:
+        //    - if file is `Unversioned` in both manifests, compare segment versions
+        //    - if file is versioned in *one* of the manifests only, compare *file* version against
+        //      other *segment* version
+        //    - if file is versioned in both manifests, simply compare file versions
+        let is_updated = old_version.or_segment_version(old.version)
+            < current_version.or_segment_version(current.version);
+
+        if is_updated {
+            updated.insert(path.clone());
+        }
+    }
+
+    updated
+}
+
+const ROCKS_DB_VIRT_FILE: &str = "__ROCKS_DB";
+const PAYLOAD_INDEX_ROCKS_DB_VIRT_FILE: &str = "__PAYLOAD_INDEX_ROCKS_DB";
```

Comment:
"Fake" files to track RocksDB version in `SegmentManifest`

Reply:
Random thought: would it be worth to use a character prefix that's even less likely to be used as file name by accident?

Like `::PAYLOAD_INDEX_ROCKS_DB`. Or do we not care?

Reply:
Good idea, we can do that.

Reply:
Done.